src/hypothesis-testing/slatkin-maddison.js

```diff
@@ -1,24 +1,84 @@
-import * as _ from "underscore";
+import { range, sum } from "d3";
+const seedrandom = require('seedrandom');
 
-function get_migration_events (tree) {
-
-
-  return 0;
 
+function cumsum(vector) {
+  var result = [];
+  vector.reduce(function(a,b,i) { return result[i] = a+b; }, 0);
+  return result;
 }
 
-/*
- * Slatkin-Maddison test
- */
-export default function slatkin_maddison(tree) {
-
-  // tree has to have an attribute
-  // 1) max parsimony, then 2) count migration events
+export default function slatkin_maddison(tree, regexes, number_of_iterations, seed) {
```

Comment:
Instead of regexes, why not just accept an object?

Reply:
In hindsight, this is definitely the appropriate separation of concerns. It also made adding CSV cleaner.